editor/editor_node.cpp

```diff
@@ -2741,47 +2741,42 @@ Dictionary EditorNode::_get_main_scene_state() {
 	state["property_edit_offset"] = get_property_editor()->get_scene_tree()->get_vscroll_bar()->get_value();
 	state["saved_version"] = saved_version;
 	state["node_filter"] = scene_tree_dock->get_filter();
+	return state;
+}
+
+void EditorNode::_set_main_scene_state(Dictionary p_state, Node *p_for_scene) {
+
 	int current = -1;
+	int index = p_state["editor_index"];
```

Comment:
It would be better to move that back where it's actually used, because:
- it's more readable
- it will crash if `!p_state.has("editor_index")`